pkg/updatechecker/updatechecker.go

```diff
@@ -207,7 +207,12 @@ func CheckForUpdates(opts CheckForUpdatesOpts) (int64, error) {
 		}
 	}()
 
-	err = store.GetStore().GetAppVersionArchive(a.ID, a.CurrentSequence, archiveDir)
+	baseSequence, err := store.GetStore().GetSequenceForLatestUpdateCursor(a.ID)
```

Comment:
How about passing `latestLicense.Spec.ChannelID` here for consistency, so that the caller knows that this is tied to a channel.